tests/conftest.py

```diff
@@ -56,3 +63,271 @@ def caplog(caplog: LogCaptureFixture):
+        bug_issue_distribution = scipy.stats.betabinom(
+            a=bug_issue_alpha, b=bug_issue_beta, n=observed_denominator
+        )
+
+        if target_lower_bound == target_upper_bound:
+            no_bug_issue_distribution = scipy.stats.binom(
+                p=target_lower_bound, n=observed_denominator
+            )
+        else:
+            a, b = self._fit_beta_distribution_to_uncertainty_interval(
+                target_lower_bound, target_upper_bound
+            )
+
+            no_bug_issue_distribution = scipy.stats.betabinom(
+                a=a, b=b, n=observed_denominator
+            )
+
+        bayes_factor = self._calculate_bayes_factor(
+            observed_numerator, bug_issue_distribution, no_bug_issue_distribution
+        )
+
+        observed_proportion = observed_numerator / observed_denominator
+        reject_null = bayes_factor > fail_bayes_factor_cutoff
+        self.proportion_test_diagnostics.append(
+            {
+                "name": name,
+                "name_addl": name_additional,
+                "observed_proportion": observed_proportion,
+                "observed_numerator": observed_numerator,
+                "observed_denominator": observed_denominator,
+                "target_lower_bound": target_lower_bound,
+                "target_upper_bound": target_upper_bound,
+                "bayes_factor": bayes_factor,
+                "reject_null": reject_null,
+            }
+        )
+
+        if reject_null:
+            if observed_proportion < target_lower_bound:
+                raise AssertionError(
+                    f"{name} value {observed_proportion:g} is significantly less than expected, bayes factor = {bayes_factor:g}"
+                )
+            else:
+                raise AssertionError(
+                    f"{name} value {observed_proportion:g} is significantly greater than expected, bayes factor = {bayes_factor:g}"
+                )
+
+        if (
+            target_lower_bound > 0
+            and self._calculate_bayes_factor(
+                0, bug_issue_distribution, no_bug_issue_distribution
+            )
+            < fail_bayes_factor_cutoff
+        ):
+            warnings.warn(
+                f"Sample size too small to ever find that the simulation's '{name}' value is less than expected."
+            )
+
+        if target_upper_bound < 1 and (
+            self._calculate_bayes_factor(
+                observed_denominator, bug_issue_distribution, no_bug_issue_distribution
+            )
+            < fail_bayes_factor_cutoff
+        ):
+            warnings.warn(
+                f"Sample size too small to ever find that the simulation's '{name}' value is greater than expected."
+            )
+
+        if fail_bayes_factor_cutoff > bayes_factor > inconclusive_bayes_factor_cutoff:
+            warnings.warn(f"Bayes factor for '{name}' is not conclusive.")
```

Comment:
The idea is that we have two cutoffs: the fail cutoff and the inconclusive cutoff. The fail cutoff is more conservative, because we want to err on the side of not forcing engineers to investigate things where nothing is broken and we were just unlucky. The inconclusive cutoff is more like a nudge to potentially run with a bigger population/sample size, but with a much higher chance of false positives, which is okay because warnings can be ignored.

Reply:
I'll be interested to learn whether any of the existing tests in pseudopeople will be inconclusive according to this metric. If not, it may not matter. But if so, we may want to run them with a bigger sample size every once in a while (assuming it is too expensive to increase the sample size we regularly run).

Reply:
Ok, seems like it's just a workflow change that we'll need to sort out, ie how do engineers let RT know when such warnings exist.

Reply:
Yeah. I think the first thing to try if you see that warning is, can I run the test with a bigger population? Then if you still get the warning with the biggest population that is feasible to run, a discussion with RT is probably in order.